pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/SelectAstNode.java

```diff
@@ -144,4 +149,27 @@ public void updateBrokerRequest(BrokerRequest brokerRequest) {
       brokerRequest.setSelections(null);
     }
   }
+
+  @Override
+  public void updatePinotQuery(PinotQuery pinotQuery) {
+    // Set data source
+    DataSource dataSource = new DataSource();
+    dataSource.setTableName(_resourceName);
+    pinotQuery.setDataSource(dataSource);
+    sendPinotQueryUpdateToChildren(pinotQuery);
+    if (_recordLimit != -1) {
```

Comment:
What happens if user specifies negative limit? We should also convert that to default value (unless we are already doing that). What do you think?

Reply:
I want to keep the behavior similar to what we have currently. This will help us in writing that generic validate/compare method. The existing code looks like this. 

```  if (selections != null) {
      if (_recordLimit != -1) {
        selections.setSize(_recordLimit);
      } else {
        // Pinot quirk: default to LIMIT 10
        selections.setSize(10);
      }
      if (_offset != -1) {
        selections.setOffset(_offset);
      }
    }```